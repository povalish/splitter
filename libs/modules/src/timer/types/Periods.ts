export enum Periods {
  Focusing = 'focusing',
  ShortBreak = 'shortbreak',
  LognBreak = 'lognbreak',
}
