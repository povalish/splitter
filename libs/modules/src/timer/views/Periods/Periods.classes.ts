import { cva } from 'class-variance-authority';

//
//

export const list = cva('flex flex-row gap-0 sm:gap-2 flex-nowrap');
