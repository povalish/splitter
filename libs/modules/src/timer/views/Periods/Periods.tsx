'use client';

import React, { useState } from 'react';
import { Tab } from '@ui-kit/components/tab/Tab';

//
//

export const Periods: React.FC = () => {
  const [selected, setSelected] = useState('tab-1');
  return (
    <ul role='tablist' className='flex gap-0 sm:gap-2 flex-nowrap'>
      <li role='presentation'>
        <Tab
          label='focusing'
          tabId='tab-1'
          panelId='panel-1'
          onSelect={() => setSelected('tab-1')}
          active={selected === 'tab-1'}
        />
      </li>

      <li role='presentation'>
        <Tab
          label='short break'
          tabId='tab-2'
          panelId='panel-2'
          onSelect={() => setSelected('tab-2')}
          active={selected === 'tab-2'}
        />
      </li>

      <li role='presentation'>
        <Tab
          label='long break'
          tabId='tab-3'
          panelId='panel-3'
          onSelect={() => setSelected('tab-3')}
          active={selected === 'tab-3'}
        />
      </li>
    </ul>
  );
}