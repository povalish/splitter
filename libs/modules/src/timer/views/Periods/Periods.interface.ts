import { Periods } from '@modules/timer/types/Periods';
import { IStyled } from '@ui-kit/types/Styled';

export interface IPeriods extends IStyled {
  active: Periods;
  onChange: (period: Periods) => void;
}
