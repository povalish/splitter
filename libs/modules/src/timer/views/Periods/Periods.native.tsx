import React from 'react';
import { View } from 'react-native';

import { Tab } from '@ui-kit/components/tab/Tab';

import { IPeriods } from './Periods.interface';
import { Periods as PeriodsEnum } from '../../types/Periods';
import { list } from './Periods.classes';

//
//

export const Periods: React.FC<IPeriods> = ({ active, className, onChange }) => (
  <View role='tablist' className={list({ className })}>
    <View role='presentation'>
      <Tab
        label='focusing'
        tabId={PeriodsEnum.Focusing}
        onSelect={() => onChange(PeriodsEnum.Focusing)}
        active={active === PeriodsEnum.Focusing}
      />
    </View>

    <View role='presentation'>
      <Tab
        label='short break'
        tabId={PeriodsEnum.ShortBreak}
        onSelect={() => onChange(PeriodsEnum.ShortBreak)}
        active={active === PeriodsEnum.ShortBreak}
      />
    </View>

    <View role='presentation'>
      <Tab
        label='long break'
        tabId={PeriodsEnum.LognBreak}
        onSelect={() => onChange(PeriodsEnum.LognBreak)}
        active={active === PeriodsEnum.LognBreak}
      />
    </View>
  </View>
);
