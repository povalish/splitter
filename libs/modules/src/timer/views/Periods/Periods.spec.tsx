import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { Periods as PeriodsComponent } from './Periods';
import { Periods } from '@modules/timer/types/Periods';

//
//

describe('TIMER MODULES - VIEWS - Periods component', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <PeriodsComponent active={Periods.Focusing} onChange={() => undefined} />,
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should call onChange with new period value', () => {
    const onChange = jest.fn();
    render(<PeriodsComponent active={Periods.Focusing} onChange={onChange} />);

    fireEvent.click(screen.getByText(/focusing/i));
    expect(onChange).toBeCalledWith(Periods.Focusing);

    fireEvent.click(screen.getByText(/short break/i));
    expect(onChange).toBeCalledWith(Periods.ShortBreak);

    fireEvent.click(screen.getByText(/long break/i));
    expect(onChange).toBeCalledWith(Periods.LognBreak);
  });
});
