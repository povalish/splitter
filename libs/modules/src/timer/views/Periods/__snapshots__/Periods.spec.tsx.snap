// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TIMER MODULES - VIEWS - Periods component should render successfully 1`] = `
<body>
  <div>
    <ul
      class="flex gap-0 sm:gap-2 flex-nowrap"
      role="tablist"
    >
      <li
        role="presentation"
      >
        <button
          aria-selected="true"
          class="
  text-[20px] text-colorText font-bold whitespace-nowrap
  md:text-[24px]
 
  py-2 px-3
  transition-all
  active:scale-95
  hover:scale-105
  focus:text-colorAccent focus:opacity-100 focus:text-shadow focus:outline-none
 opacity-100"
          id="focusing"
          role="tab"
          type="button"
        >
          focusing
        </button>
      </li>
      <li
        role="presentation"
      >
        <button
          aria-selected="false"
          class="
  text-[20px] text-colorText font-bold whitespace-nowrap
  md:text-[24px]
 
  py-2 px-3
  transition-all
  active:scale-95
  hover:scale-105
  focus:text-colorAccent focus:opacity-100 focus:text-shadow focus:outline-none
 opacity-50"
          id="shortbreak"
          role="tab"
          type="button"
        >
          short break
        </button>
      </li>
      <li
        role="presentation"
      >
        <button
          aria-selected="false"
          class="
  text-[20px] text-colorText font-bold whitespace-nowrap
  md:text-[24px]
 
  py-2 px-3
  transition-all
  active:scale-95
  hover:scale-105
  focus:text-colorAccent focus:opacity-100 focus:text-shadow focus:outline-none
 opacity-50"
          id="lognbreak"
          role="tab"
          type="button"
        >
          long break
        </button>
      </li>
    </ul>
  </div>
</body>
`;
